Bottom: 595d16b81a0c4e92b953604054118d2e7544a3e5
Top:    a1d9ab0d1d177ca7fdf5336aa9b00ea760c62ba3
Author: Ian Kent <raven@themaw.net>
Date:   2013-03-21 12:07:41 +0800

Refresh of autofs-5.0.5-wait-for-master-map-available-at-start.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index 293d8dc..4f5d0ca 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1899,12 +1899,12 @@ static void remove_empty_args(char **argv, int *argc)
 	*argc = j;
 }
 
-static int do_master_read_master(struct master *master, time_t age, unsigned int wait)
+static int do_master_read_master(struct master *master, time_t age, int wait)
 {
 	sigset_t signalset;
-	unsigned int max_wait = wait;
 	unsigned int retry_wait = 2;
 	unsigned int elapsed = 0;
+	int max_wait = wait;
 	int ret = 0;
 
 	sigemptyset(&signalset);
@@ -1925,7 +1925,7 @@ static int do_master_read_master(struct master *master, time_t age, unsigned int
 			break;
 
 		elapsed += retry_wait;
-		if (elapsed >= max_wait) {
+		if (max_wait > 0 && elapsed >= max_wait) {
 			logmsg("problem reading master map, "
 				"maximum wait exceeded");
 			break;
@@ -2305,8 +2305,8 @@ int main(int argc, char *argv[])
 		 * a signal is received, in which case exit returning an
 		 * error.
 		 */
-		if (!do_master_read_master(master_list, age, 180)) {
-			logerr("%s: failed to read master map after 180 seconds!",
+		if (!do_master_read_master(master_list, age, -1)) {
+			logerr("%s: failed to read master map!",
 			       program);
 			master_kill(master_list);
 			release_flag_file();
